### ISSUE TITLE
MIXER - Length Of Channel Labels

### ISSUE THREAD

**Issue by miketurn** [LMMS/lmms#3156]
Hello Everyone,

Probably not a huge deal to most people, but after adding the "send" and send knobs to the mixer (which was a great addtion well worth the sacrafice) it cut the length available for the channel labels down quite a bit.  Might be nice to someday see a little more room given for them.
Also I have mentioned this in the past, but it also sometimes looks a little bit like a mistake when the label goes behind the channel number (I was told this was intentional but it still makes me uncomfortable).  Having a little bit more extra room for labels would prevent that from happen too often.

I admit I love the look/simplicity of the MIXER and wouldn't want to see it change a great deal, but on the other hand getting some desired features to be added would enhance its functionality.  Removing as much of its padding and adding a little height to the MIXER itself is probably the solution (the width of each channel is perfect).  This would give room for the upcoming "solo" button, a little extra label length, and the pan knobs that only I want but nobody else does.  :)

Anyways if anyone else agrees with a little extra label room , please share your opinions.

**Comment by BaraMGB** [comment]
>Also I have mentioned this in the past, but it also sometimes looks a little bit like a mistake when the label goes behind the channel number (I was told this was intentional but it still makes me uncomfortable). 

This is fixed in 1.2 . 

We have to consider more room for the Mixer means lesser room for other windows. At the moment more room were only for the labels. I guess the current heigth is a good compromise for small displays.

**Comment by musikBear** [comment]
just a thought
Does the whole instrument name need to be written in FX-mixer?
If the name-label in FX-mixer was limited to always max ..8 letters, then the mixer would look tidy, and i think the needed info is presented with 8 letters.
Mouseover could still show the whole name, as tool-tip, or better, when high-lighted: in a dedicated info-display, on main-component, to the right of the CPU/dB display. There is a lot of space there.
Mockup:
![mainwinfo](https://cloud.githubusercontent.com/assets/6662066/21319875/3c4d1cb0-c60f-11e6-894c-a46b9b07f463.PNG)

**Comment by miketurn** [comment]
@musikBear 
8 Letters! No musikBear, please No   :)
This reminds me of the days of DOS filenames where everything had to be abbreviated.

Personally, I feel that the smaller the labels are, the more one would have to think about what the abbreviations are representing.  Things like small abbreviated labels, and having to look to other locations to see if items have been panned (any chance I get to throw in my opinion of getting pan knobs in the mixer I will take it), etc. I feel would all take from the person who is mixings focus away from the mixing process.  Any time you are forced to think of something else at that time the less focus you are placing on listening and adjusting what is needed.
Personally I like to put a prefix to the label stating what it is as well, for example
Drums - Kick
Drums - Snare
Guitar - Chords
Guitar - Solo
Without the ability to change the colors of mixer channels, I feel it helps to keep everything organized and less confusing.

The "Info" bar could however have some uses, this in my mind is similar to the one found in FL Studio where it gives newer users a desciption of what items are when hovering over them.  This feature may be nice to add, but I would say if it was added to make it be able to be hidden if not desired.

**Comment by miketurn** [comment]
I was able to get 1.2.0 RC1 to work on a labtop and saw that the "solo" button has been added to the mixer.  The beginning of the labels are so close to the red "solo" button it looks like a mistake, they should probably be moved away from them just a tad.  I know it is a GUI issue and has no affect on the mixer itself but it just doesn't look right, in my opinion.

@BaraMGB
I tested the label cutoff issue that you mention was fixed in 1.2.0 but I still experienced the same result, does it not do this on your machine?

**Comment by BaraMGB** [comment]
@miketurn please provide a screen shot.

**Comment by miketurn** [comment]
@BaraMGB 
Thank You for your response.
A couple of weird things I notice in the picture you provide.

1.) First off I actually like the skin you are using, is that available somewhere or did you create that yourself?  I did the same thing with making the effects background just a color rectangle.

2.) I mention the solo buttons getting too close to the label, but in yours it looks okay, maybe because your skin eliminates the border around the buttons, but even still, it seems you have more space there.

3.) I see how your labels have the ....    wow that seems like it cuts the label off quite a bit before the channel number graphic, look how tiny those labels have to be, but anyways yeah yours does have cut off. 

4.) In the picture you provided, it appears as if you are editing the label of channel 22, odd because it looks like you are typing vertically instead of getting a dialog box like I get.

I just noticed that RC2 has been released recently, I just downloaded it, when testing I was using RC1 because RC2 was not available, I will try this release out and see if maybe it will work for me in RC2.

5.) Is the picture you provide using release RC1 or RC2?

Thank You

**Comment by musikBear** [comment]
@BaraMGB 
> Fx-mixer looks like this atm:

It fine, but look at this (in 1.1.3 layout)
![fxmixer-preset-area](https://cloud.githubusercontent.com/assets/6662066/21814564/fa049c96-d759-11e6-8464-9de90fc2dc4a.PNG)
With that added, the issue of either vertical/ truncated/  names, would be gone. It ought to be as simple as a concatenation to header (if codebase was as3 :)

**Comment by Umcaruje** [comment]
@musikBear that's actually not a bad idea, we have a lot of vertical realestate on the title bar and we should use it. Did you make an enhancement request for that?

**Comment by miketurn** [comment]
Ahhh, I don't know about that, personally I don't like the look of the channel name up in the title bar.
To me it looks like it doesn't belong there, makes the text off center, would kind of clutter up the clean new look implemented in RC2.  
I have an idea that I have shared, on another forum, and seeing an odd possible graphical bug in 1.2.0 RC2 that I have found, shows it can be done.  I will explain later once I have a little more time to think about it.
